syntax = "proto3";

package interactor;

option go_package = "github.com/trading-bot/stock-interactor/internal/api/pb;pb";

service HistoryService {
    rpc GetHistory(HistoryRequest) returns (stream HistoryResponse);
}

message HistoryRequest {
    string symbol = 1;
    string interval = 2;
    int64 from_timestamp = 3;
    int64 to_timestamp = 4;
}
message HistoryResponse {
    repeated Candle candles = 1;
}

message Candle {
  double open_time = 1;         // Время открытия (мс)
  double open = 2;             // Цена открытия
  double high = 3;             // Максимальная цена
  double low = 4;              // Минимальная цена
  double close = 5;            // Цена закрытия
  double volume = 6;           // Объём в базовом активе
  double close_time = 7;        // Время закрытия (мс)
  double quote_volume = 8;     // Объём в котируемом активе
  int32 trade_count = 9;       // Количество сделок
  double taker_buy_base = 10;  // Объём покупок у мейкеров (в base asset)
  double taker_buy_quote = 11; // Объём покупок у мейкеров (в quote asset)
}